codex/stores/repostore.nim

```diff
@@ -274,17 +389,37 @@ method ensureExpiry*(
   ## Ensure that block's associated expiry is at least given timestamp
   ## If the current expiry is lower then it is updated to the given one, otherwise it is left intact
   ##
-  without cidAndProof =? await self.getCidAndProof(treeCid, index), err:
+
+  without leafMd =? await self.getLeafMetadata(treeCid, index), err:
     return failure(err)
 
-  await self.ensureExpiry(cidAndProof[0], expiry)
+  await self.ensureExpiry(leafMd.blkCid, expiry)
+
+method putBlockCidAndProof*(
+  self: RepoStore,
+  treeCid: Cid,
+  index: Natural,
+  blkCid: Cid,
+  proof: MerkleProof
+): Future[?!void] {.async.} =
+  ## Put a block to the blockstore
+  ##
+
+  logScope:
+    treeCid = treeCid
+    index = index
+    blkCid = blkCid
 
-proc persistTotalBlocksCount(self: RepoStore): Future[?!void] {.async.} =
-  if err =? (await self.metaDs.put(
-      CodexTotalBlocksKey,
-      @(self.totalBlocks.uint64.toBytesBE))).errorOption:
-    trace "Error total blocks key!", err = err.msg
+  without res =? await self.putLeafMetadata(treeCid, index, blkCid, proof), err:
     return failure(err)
+
+  if res == Stored:
+    if err =? (await self.updateBlockMetadata(blkCid, plusRefCount = 1)).errorOption:
+      return failure(err)
+    trace "Leaf metadata stored, block refCount incremented"
+  else:
+    trace "Leaf metadata already exists"
```

Comment:
No, the refCount refers to how many trees are referring this block.

If you store just a block (without any references to the tree yet), what should be the initial refCount for this block? In my opinion it's `0` - since there are no references to it.

Then when adding new references to this block (new leaf metadata) we increase the refCount by 1.